### PR TITLE
Finalize chapter 7 on the superbuild pattern

### PR DESCRIPTION
**Pull request by robertodr**

### CODE REVIEW

**Comment by robertodr** [comment]
All recipes in place. Let's see what Travis has to say about them and then I'll squash into one commit.

**Comment by bast** [comment]
Holy superbuilds, batman!

File: Chapter07/recipe-03/c-example/src/CMakeLists.txt [hunk 1]
```diff
@@ -1,12 +1,14 @@
...
-project(fftw_example C)
+project(recipe-03_core C)
 
 set(CMAKE_C_STANDARD 99)
 set(CMAKE_C_EXTENSIONS OFF)
 set(CMAKE_C_STANDARD_REQUIRED ON)
 
 find_package(FFTW3 CONFIG REQUIRED)
+get_property(_loc TARGET FFTW3::fftw3 PROPERTY LOCATION)
+message(STATUS "Found FFTW3: ${_loc} (found version ${FFTW3_VERSION})")
```
Comment (bast) [inline Chapter07/recipe-03/c-example/src/CMakeLists.txt]: nice!

File: Chapter07/recipe-04/cxx-example/src/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  PUBLIC
+    Eigen3::Eigen
+    OpenMP::OpenMP_CXX
+  )
+
+install(
+  PROGRAMS
+    ${CMAKE_BINARY_DIR}/linear-algebra
+  DESTINATION
+    ${CMAKE_INSTALL_PREFIX}/bin
```
Comment (bast) [inline Chapter07/recipe-04/cxx-example/src/CMakeLists.txt]: Should we here go for the gnuinstalldirs?

Comment (robertodr) [reply Chapter07/recipe-04/cxx-example/src/CMakeLists.txt]: We only mention that in chapter 10. I think here it's fine to "do it by hand" and then have an hooplà moment in chapter 10.

Comment (bast) [reply Chapter07/recipe-04/cxx-example/src/CMakeLists.txt]: Cool :-)

**Comment by robertodr** [comment]
Not too bad on Travis. HDF5 superbuild might time us out now and then. And it's not Ninja tolerant, so I need to skip it.

**Comment by robertodr** [comment]
Some of the superbuilds have to be backported to CMake 3.5 Working on it.

**Comment by robertodr** [comment]
This is IMO ready to go. I've fixed the CI as much as possible. Sadly, we've started to time out on Travis (despite the dependency caching)

**Comment by bast** [comment]
Oops, also conflict in chapter 8. Please let me know if I should look at it. I will approve the changes and as soon as you are happy with test results, please merge.

**Comment by robertodr** [comment]
I've rebased. I'll look at CI later today.

**Comment by bast** [comment]
Great. It might be that test breakage is not due to chapter 7. I have merged chapter 8 although there was red stuff but before that I have verified that testing only chapter 8 is all green on Travis. We might need to work around timeouts and speed things up but this is outside of scope chapter 7.

**Comment by robertodr** [comment]
One of the conda recipes times out, but the recipes in this chapter are (finally!) fine. I am merging.

**Pull request merged**